doc/tutorial/energe_lstmbasedtimeseries.md

```diff
@@ -0,0 +1,283 @@
+
+The dataset is taken from the GEFCom2014 energy forecasting [competition](Tao Hong, Pierre Pinson, Shu Fan, Hamidreza Zareipour, Alberto Troccoli and Rob J. Hyndman, "Probabilistic energy forecasting: Global Energy Forecasting Competition 2014 and beyond", International Journal of Forecasting, vol.32, no.3, pp 896-913, July-September, 2016.). It consists of years of hourly electricity load data from the New England ISO and also includes hourly temperature data. We just choose part of electricity load data which date range is from 2014-10-01 to 2014-12-31 to complete our tutorial. And, we load the dataset into MySQL manually.
+
+In this notebook, we will demonstrate how to:
+
+- Prepare time series data for training an LSTM forecasting model with SQL
+  - Data scaling
+  - Reconstruct series data
+  - Split the raw data into the train set and test set
+- Train an LSTMBasedTimeSeriesModel on the dataset.
+- Predict one or more time-step ahead electricity load data, using historical load data only. 
+
+## PART 1  Prepare Data
+
+The energy data set contains two features, one is the date-time column, and the other is electricity load data. 
+
+We can have a quick peek of the raw data by running the following standard SQL statements. 
+
+```sql
+%%sqlflow
+select * from energy.raw limit 10;
+```
+
+```sql
+%%sqlflow
+select count(*) from energy.raw;
+```
+
+### PART 1.1  Scale Data
+
+Because the raw data can't be used directly to build the model, we need to reconstruct the raw data into the series data instead. Before that, we need to scale the data first into the range of (0, 1). It should be noted that data scaling must be done before the data reconstruct shown in the next step.
+
+```sql
+%%sqlflow
+drop table if exists energy.normalized;
+create table energy.normalized as
+select
+    dt,
+    (
+        energy - (select min(ta.energy) as a from energy.raw as ta)
+    ) / (
+        (select max(ta.energy) as a from energy.raw as ta)  
+        -(select min(ta.energy) as a from energy.raw as ta)
+    ) as energy
+from
+    energy.raw;
+```
+
+Let's have a quick peek of the normalized data.
+
+```sql
+%%sqlflow
+select * from energy.normalized limit 5;
+```
+
+### PART 1.2  Reconstruct Data
+
+Then, you need to choose an appropriate timestep of the series (n_in) and you need to specify how much time-steps target data you want to predict (n_out) by the trained model. By the way, the `n_in` parameter can be tuned manually during the model training stage.
+
+As the following shows, because we want to train a model that `n_in = 10 `and `n_out=4` in the next stage, we construct a time series data that length is 14. The reconstruct MySQL Statement is:
+
+```sql
+%%sqlflow
+drop table if exists energy.con_all;
+create table energy.con_all as
+select * from(
+select
+    t.dt,
+    @lagfield0:= @lagfield1 as col_1,
```

Comment:
Add a simple explanation for the syntax?

Reply:
OK~ Thanks~~